src/angular/src/app/spark-core-angular/directives/inputs/sprk-input/sprk-input.directive.ts

```diff
@@ -0,0 +1,43 @@
+import { Directive, Input, OnInit, OnChanges, ElementRef } from '@angular/core';
+
+@Directive({
+  selector: '[sprk-input]'
+})
+
+export class SprkInputDirective implements OnInit, OnChanges {
+  @Input() additionalClasses: string;
+  @Input() valid: boolean = true;
+
+  constructor(public ref: ElementRef){};
+
+  getClasses(): Array<String> {
+    let classArray: Array<String> = [];
+
+    if(this.ref.nativeElement.nodeName === 'SELECT') {
```

Comment:
space needed here if(this

Reply:
why isn't prettier or something catching these?

Reply:
We don't have it setup yet.